internal/security/credentials_test.go

```diff
@@ -136,13 +175,82 @@ func TestGetSecrets(t *testing.T) {
 
 		// not re-newing the secretProvider will test the cache for the next item in the getSecretsTestData slice
 		if test.resetSecretsCache {
-			secretProvider = newMockSecretProvider()
+			secretProvider = newMockSecretProvider(nil)
 		}
 	}
 }
 
-func newMockSecretProvider() *SecretProvider {
-	return &SecretProvider{secretClient: &mockSecretClient{}, secrets: make(map[string]map[string]string)}
+func TestGetInsecureSecrets(t *testing.T) {
+
+	secretProvider, origEnv := setupGetInsecureSecrets(t)
+
+	for _, test := range getSecretsTestData {
+		testNameInfo := fmt.Sprintf("Test: %v", test.testName)
+
+		path := test.path
+		keys := test.keys
+
+		secrets, err := secretProvider.getInsecureSecrets(path, keys...)
+
+		assert.Equal(t, test.expectedErr, err, testNameInfo)
```

Comment:
Use `require.Equal()` so failure is Fatal, since no need to continue test. This requires you to make these true table driven.